doc/adr/0009-container-image-tags.md

```diff
@@ -0,0 +1,55 @@
+# 9. Convention for tagging of container images
+
+Date: 2020-03-13
+
+## Status
+
+Proposed
+
+## Context
+
+Libero's products are assembled from product specific, general libero and external container images.
+
+Integration testing and reproducible deploys require uniquely identifiable image tags.
+
+
+## Decision
+
+### Tagging flow
+
+- each container image of a Libero container shall be tagged with the git-sha of the build
+- compose files used for integration test shall use immutable image tags
+- the compose files will track the component images and external images with the help of PR generating bots (e.g. dependabot, renovate)
+- upon passed integration tests the used libero component images get tagged with the git-sha of the integration/umbrella repo
+- semver tagging of Libero component images is triggered by tagging of the umbrelle repo
+
+### Tag Structure
+
+Component build passes tests and gets pushed to registry:
+
+- liberoadmin/reviewer-client:full_git_hash
```

Comment:
Should we consider also pushing an additional tag with the semver here if building from a tag? e.g.

docker build --no-cache -t liberoadmin/editor-client:$GITHUB_SHA -t liberoadmin/editor-client:1.0.2 .

I currently version each of my components seperately.

Reply:
We could do both. Then the integration/umbrella repo could have two compose files: one tracking minor/bug changes of the components, the other tracking "latest/unstable".

Do you coordinate the versions of your components in some way e.g. 1.x will always work with other 1.x components?

Reply:
This should be the case ... we are still not thinking about versioning. My vote would be to go for the double-tagging - this gives us flexibility in the future... with the expectation of moving to semver.